GUI/SearchTaskWindow.xaml.cs

```diff
@@ -524,7 +524,8 @@ private void SaveButton_Click(object sender, RoutedEventArgs e)
                 TheTask.SearchParameters.CustomMdac = customkMdacTextBox.Text;
             }
 
-            //determine if semi or nonspecific with a specific protease. Full is already added by default.
+            //determine if semi or nonspecific with a specific protease.
+            TheTask.SearchParameters.LocalFdrCategories = new List<FdrCategory> { FdrCategory.FullySpecific }; //set the default again to reset the list
             if(searchModeType == CleavageSpecificity.Semi || protease.CleavageSpecificity==CleavageSpecificity.Semi)
```

Comment:
i'd like to change this structure a bit. instead of adding to a list, can we just set the list based on conditions? seems less error-prone.
```
if(semispecific)
{
   SearchParameters.LocalFdrCategories = new List<>{ Full, Semi };
}
if(nonspecific)
{
   SearchParameters.LocalFdrCategories = new List<>{ Semi, Non };
}
else
{
   SearchParameters.LocalFdrCategories = new List<>{ Full };
}
```

Also, it seems weird to me to have semi-specific FDR category as a separate FDR category for the nonspecific search. If I'm doing a "true" nonspecific search and there is no digestion motif at all, what does "semispecific" even mean?

Reply:
I'll make the first change.
You can do nonspecific searches of specific proteases  (rare in trypsin, common in chymotrypsin). In that case, full, semi, and non are all stratified.

Reply:
ok, so Non-specific search mode with "chymotrypsin" selected as the protease is a valid option?

what happens if I select "Non-specific" as the protease with Non-specific search mode checked? Is there only effectively one FDR category then?

Reply:
Correct.
If you do non-specific with non-specific, there are still three categories, but only the non-specific one ever gets used.